tests/topotests/multicast_pim_dr_nondr_test/test_multicast_pim_dr_nondr_topo2.py

```diff
@@ -0,0 +1,1126 @@
+#!/usr/bin/env python
+
+#
+# Copyright (c) 2020 by VMware, Inc. ("VMware")
+# Used Copyright (c) 2018 by Network Device Education Foundation,
+# Inc. ("NetDEF") in this file.
+#
+# Permission to use, copy, modify, and/or distribute this software
+# for any purpose with or without fee is hereby granted, provided
+# that the above copyright notice and this permission notice appear
+# in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND VMWARE DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL VMWARE BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+Following tests are covered to test multicast pim sm:
+
+Test steps
+- Create topology (setup module)
+- Bring up topology
+
+Following tests are covered:
+1. Configure IGMP local join on DR and non DR
+"""
+
+import os
+import sys
+import json
+import time
+import datetime
+from time import sleep
+import pytest
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# Required to instantiate the topology builder class.
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    write_test_footer,
+    step,
+    reset_config_on_routers,
+    shutdown_bringup_interface,
+    apply_raw_config,
+    add_interfaces_to_vlan,
+    kill_router_daemons,
+    start_router_daemons,
+    create_static_routes,
+    check_router_status,
+    topo_daemons,
+    required_linux_kernel_version,
+)
+from lib.pim import (
+    create_pim_config,
+    create_igmp_config,
+    verify_ip_mroutes,
+    clear_ip_mroute,
+    clear_ip_pim_interface_traffic,
+    verify_pim_config,
+    verify_upstream_iif,
+    verify_multicast_traffic,
+    verify_multicast_flag_state,
+    verify_igmp_groups,
+    McastTesterHelper
+)
+from lib.topolog import logger
+from lib.topojson import build_config_from_json
+
+HELLO_TIMER = 1
+HOLD_TIMER = 3
+
+TOPOLOGY = """
+
+         r5 --- i2
+         |
+         |
+     --- r4 ----
+    |           |
+    |           |
+    r1 -- s1 -- r2
+          |
+          |
+    i1 -- r3
+
+    Description:
+    i1, i2  - FRR running iperf to send IGMP
```

Comment:
how i1,i2 are running iperf?

Reply:
@ton31337 , here i1, i2 are used to send IGMP join and traffic using iperf commands.